pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java

```diff
@@ -110,6 +113,8 @@ public static void createHelixClusterIfNeeded(String helixClusterName, String zk
           zkClient.close();
         }
       }
+      createLeadControllerResourceIfNeeded(helixClusterName, admin, enableBatchMessageMode,
```

Comment:
When we already have a cluster and everything is good, we will return from line 100, right? So, we will never create controller resource

Reply:
Good catch! I'll add that after Line 100.

Reply:
I created a new method called `setupHelixCluster()` which will call `createHelixClusterIfNeeded` and `createLeadControllerResourceIfNeeded` sequentially.